### PR TITLE
add delete and duplicate row buttons to expanded form and grid view

### PR DESCRIPTION
**Pull request by spaudanjo**
## Change Summary

Contributes to https://github.com/nocodb/nocodb/issues/2101 and adds also a 'Delete row' and 'Duplicate row' menu to the expanded form view. 

re #435

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/expanded-form/index.vue [hunk 1]
```diff
@@ -235,11 +258,13 @@ export default {
 
 .nc-prev-arrow,
 .nc-next-arrow {
-  @apply absolute opacity-70 rounded-full transition-transform transition-background transition-opacity transform bg-white hover:(bg-gray-200) active:(scale-125 opacity-100) text-xl;
+  @apply absolute opacity-70 rounded-full transition-transform transition-background transition-opacity transform bg-white hover: (bg-gray-200) active:(scale-125 opacity-100) text-xl;
```
Comment (pranavxc) [inline packages/nc-gui/components/smartsheet/expanded-form/index.vue:261]: `hover: (bg-gray-200)` => `hover:(bg-gray-200)`

**Comment by pranavxc** [comment]
re #2101

**Review by pranavxc — APPROVED** [review]

**Review by dstala — CHANGES_REQUESTED** [review]
https://www.loom.com/share/08264aab38bd4bffb65510de7eac9518

in fact, when a row is duplicated - we can disable both `duplicate row` and `delete row` option un less its saved

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 2]
```diff
@@ -168,85 +172,134 @@ watch(view, async (nextView) => {
...
+          <a-menu-item v-if="contextMenuTarget" @click="openNewRecordFormHook.trigger()">
+            <div v-e="['a:row:insert']" class="nc-project-menu-item">
+              <!-- Insert New Row -->
+              {{ $t('activity.insertRow') }}
+            </div>
+          </a-menu-item>
+        </a-menu>
+      </template>
+      <div class="nc-gallery-container grid gap-2 my-4 px-3">
+        <!-- v-for="(row, rowIndex) of data -->
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Gallery.vue:226]: can be removed

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 3]
```diff
@@ -81,6 +82,9 @@ const isRowEmpty = (record: any, col: any) => {
   return Array.isArray(val) && val.length === 0
 }
 
+// const { xWhere, isPkAvail, isSqlView, eventBus } = useSmartsheetStoreOrThrow()
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Gallery.vue:85]: can be removed

File: packages/nc-gui/components/smartsheet/expanded-form/index.vue [hunk 4]
```diff
@@ -148,7 +169,7 @@ export default {
     class="nc-drawer-expanded-form"
     :class="{ active: isExpanded }"
   >
-    <SmartsheetExpandedFormHeader :view="props.view" @cancel="onClose" />
+    <SmartsheetExpandedFormHeader :view="props.view" @cancel="onClose" @duplicateRow="onDuplicateRow" />
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/expanded-form/index.vue:172]: v-on event `@duplicateRow` must be hyphenated

**Review by wingkwong — APPROVED** [review]

**Comment by spaudanjo** [comment]
the tests lgtm :)

**Review by dstala — APPROVED** [review]

**Pull request merged**